storage/innobase/log/log0recv.cc

```diff
@@ -3785,6 +3785,7 @@ void recv_sys_t::apply(bool last_batch)
       /* We skipped this in buf_page_create(). */
       mlog_init.mark_ibuf_exist();
     mlog_init.clear();
+    dblwr.pages.clear();
```

Comment:
Is there any reason why we couldn’t assign `recovery_on= false` here?

Reply:
In case of log file resizing during restart, InnoDB expects the `recv_sys.recovery_on` to be true in `srv_prepare_to_delete_redo_log_file()`

Reply:
Based on the assertion failure during log resizing in https://buildbot.mariadb.org/#/builders/640/builds/8172/steps/7/logs/stdio
it looks like we need `recv_sys.recovery_on=false` in order to prevent any log to be written after a page flush batch has completed. So, we can’t add the assignment here. The assignment at a later point needs to be accompanied with a comment that explains this.

Reply:
Sorry, I meant that we need `recv_sys.recovery_on=true` to disable any checkpoint related log writes in `buf_flush_page_cleaner`. If we are rebuilding the redo log during startup not because of a size change, but because of a format change, we want to avoid any log corruption (6e5333fc8cf9efaee2abc7e34982a862c2268284) due to writing new-format log records to an old log file.